packages/hardhat-zksync-solc/src/constants.ts

```diff
@@ -0,0 +1,28 @@
+import { ZkSolcConfig } from './types';
+
+export const ZK_ARTIFACT_FORMAT_VERSION = 'hh-zksolc-artifact-1';
+export const LATEST_VERSION = '1.2.1';
+
+export const defaultZkSolcConfig: ZkSolcConfig = {
+    version: LATEST_VERSION,
+    compilerSource: 'binary',
+    settings: {
+        compilerPath: '',
+        experimental: {},
+    },
+};
+
+export const SUPPORTED_ZKSOLC_CONTRACT_OUTPUT_SELECTIONS = [
+    'abi',
+    'evm.methodIdentifiers',
+    'storageLayout',
+    'irOptimized',
+    'evm.legacyAssembly',
+];
+
+export const defaultSolcOutputSelectionConfig = {
+    '*': {
+        '*': ['abi', 'evm.methodIdentifiers'],
+        '': ['ast'],
```

Comment:
Do we actually want to include the `ast` for all contracts by default? The artifacts would be huge

Reply:
Our logic here was to include all supported output selections that are also added by hardhat core (https://github.com/NomicFoundation/hardhat/blob/7fc76390b5a1b2b25877026466f295f29ff95b6c/packages/hardhat-core/src/internal/core/config/default-config.ts#L139) and 'ast' is one of them. Also, I believe this field is needed for supporting upgradable contracts, so maybe it's useful to keep it as a default...

Reply:
I agree. Optional fields must only be requested by the user in hardhat.config.
Although zksolc will request AST anyway for its internal use, always having this field here by default will certainly be unwanted.

Reply:
Ok, if it can be breaking, let's keep it. Perhaps, I'll remove it from the zksolc defaults one day.